store/rootmulti/store.go

```diff
@@ -485,12 +515,14 @@ func (rs *Store) CacheWrapWithTrace(_ io.Writer, _ types.TraceContext) types.Cac
 // CacheMultiStore creates ephemeral branch of the multi-store and returns a CacheMultiStore.
 // It implements the MultiStore interface.
 func (rs *Store) CacheMultiStore() types.CacheMultiStore {
+	rs.listenersMx.Lock()
+	defer rs.listenersMx.Unlock()
 	stores := make(map[types.StoreKey]types.CacheWrapper)
 	for k, v := range rs.stores {
 		store := types.KVStore(v)
 		// Wire the listenkv.Store to allow listeners to observe the writes from the cache store,
 		// set same listeners on cache store will observe duplicated writes.
-		if rs.ListeningEnabled(k) {
+		if rs.listeningEnabled(k) {
```

Comment:
This is an API breaking change. Why was it made?

Reply:
This is part of the `listenerMx` conversation. ref: https://github.com/cosmos/cosmos-sdk/pull/14207#discussion_r1096560673.

Reply:
This breaks the API w/o sufficient reason. I don't understand the convo around concurrency.

Reply:
reverted